### DISCUSSION TITLE
Create a new named buffer

### DISCUSSION THREAD

**Discussion by GlassGruber** [helix-editor/helix#5865] [Q&A]
I see that if I use the `:new` command I can't set a name for the new scratch buffer created. Hope this is not an RTFM.

As a side note I also noticed that I can't create a new buffer if empty, I must write something before being able to create a new scratch buffer.

Thank you!

**Comment by CptPotato** [comment]
I actually ran into this quite a few times, too. I expected that `:new foo.rs` and `:w` would write the file to the file name I initially specified but it actually doesn't.

I think this would be a good addition. This also makes it more consistent with opening helix with a new file name: `$ hx foo.rs`, `:w` for example works. I didn't find an existing issue that mentions this; you can create a new `enhancement` issue for this if you want.

**Reply by GlassGruber** [reply]
> I didn't find an existing issue that mentions this; you can create a new enhancement issue for this if you want.

Sure! Do I just copy paste what I wrote? Should I add your comment too?

**Reply by CptPotato** [reply]
Hmm, I just noticed that `:open xyz` actually works as I would expect `:new xyz` to function. i.e. `:o foo.txt bar.rs` will create two buffers with the corresponding file name.

> Sure! Do I just copy paste what I wrote? Should I add your comment too?

I've already gone ahead and created one(#5867).

(You can create an issue from a discussion in the side bar `Create issue from discussion`, it will just copy paste the original post and link the discussion. Alternatively you can create a new issue and link it yourself using it's id `#5865`).

**Reply by GlassGruber** [reply]
Ha! Incidentally I just discovered what you replied and was about to report same thing! 🙃 

The problem is that maybe I've found another minor issue: I was trying to do this `:o %SOME_PATHENV%\somefile.txt`, but I got an empty buffer named as `somefile.txt` instead of a file I knew it exists at a long path stored in `%SOME_PATHENV%`.

Is it possible to use an ENV variable in command palette? Should I open a new discussion maybe?

> I've already gone ahead and created one(https://github.com/helix-editor/helix/issues/5867).
> 
> (You can create an issue from a discussion in the side bar Create issue from discussion, it will just copy paste the original post and link the discussion. Alternatively you can create a new issue and link it yourself using it's id #5865).

Ah interesting noticed just now, thank you!
